types/math/dec.go

```diff
@@ -35,6 +35,28 @@ func NewDecFromString(s string) (Dec, error) {
 	return Dec{*d}, nil
 }
 
+func NewNonNegativeDecFromString(s string) (Dec, error) {
+	d, err := NewDecFromString(s)
+	if err != nil {
+		return Dec{}, err
+	}
+	if d.IsNegative() {
+		return Dec{}, fmt.Errorf("cannot parse non negative decimal: %s", d.String())
+	}
+	return d, nil
+}
+
+func NewPositiveFixedDecFromString(s string, max uint32) (Dec, error) {
```

Comment:
I think we actually need two functions, `NewNonNegativeFixedDecFromString` and `NewPositiveFixedDecFromString`, as some of the use cases below allow zeroes and some don't.

Also this function is allowing zero currently, so not quite returning positive values!

Reply:
[added](https://github.com/regen-network/regen-ledger/pull/452/commits/f72b7d629550f348ceb8e0262b822b353df5a709#diff-f0b0d64c76c90877cf0c69b29e675ed4129457dfbfdede76b2d278eabb8e5c25R49-R62)